tests/e2e/helpers/loadExtension.ts

```diff
@@ -1,16 +1,35 @@
 import puppeteer from "puppeteer";
+import fs from 'fs';
 
-const delay = async (time) => {
+export const delay = async (time) => {
   return new Promise(function (resolve) {
     setTimeout(resolve, time);
   });
 };
 
-export const loadExtension = async () => {
-  const extensionPath = process.env.CI
+async function fileExists(f){
+  try {
+      await fs.promises.access(f, fs.constants.F_OK)
+      return true;
+  } catch (e) {
+      return false;
+  }
+}
+
+/**
+ * Load the extension
+ * @param production specify if you want to load the production or development mode
+ */
+export const loadExtension = async (production?) => {
+  if(typeof production=="undefined")production=process.env.CI;
```

Comment:
Uhm, I get what you are doing here but it's confusing to me.  
Is there any way, we could make this simpler?  

I believe tests should run against a prod version of the code. In this case we need to "inject" special test relevant mocking code.
Is it possible to i.e. do a production build but inject that code as well? Maybe instead of using `process.env.NODE_ENV` something like `process.env.MOCK_E2E_CODE`. Just an idea.

Reply:
That would make a third build that is an unmarked production build with exposed apis, i think it would be more confusing.

Reply:
What do you mean with "unmarked"?  
The reason I prefer prod builds for E2E testing is that some issues only appear once the code is built.

Reply:
Only testBatteries.spec.ts will run in development mode, i think it is a decent compromise, the extension is tested in production while the batteries in development mode. 

If we want to make things proper we should have a third build that is for *testing*  with production code+testing hooks, but since it requires some extensive change to the build script, i think it should probably go in its own PR, and then "testing" can be used instead of "development" . 
With unmarked i mean there would be two types of production builds, the one that is really production and the one with injected code and they will be both marked as a production, i fear this might lead to testing code being leaked into code that is expected to be production only.

Reply:
Got it. I'm ok with rethinking this in a later PR if the others are ok with it.  
If we create a new npm-script to create a prod build with injected mock-code I'm ok with that. Only `yarn build` will ever be used for the acutal published extension.